From a23d2257c59c6b288fff63ebf5678ebd252c40e3 Mon Sep 17 00:00:00 2001
From: James Slagle <jslagle@redhat.com>
Date: Thu, 6 Mar 2014 15:38:16 -0800
Subject: [PATCH] os-*-config package install support

Package install support for:
* os-collect-config
* os-refresh-config
* os-apply-config

This change requires a corresponding change to dib to move the install
type enablement out of source-repositories since the os-*-config
packages do not use source-repositories:
cccdcb904d485fce92443edbb088740e074c3046

Change-Id: I22672a4312152ab594de2a149e16866c7564e3d4
---
 .../os-apply-config/install.d/10-os-config-applier | 19 --------
 .../10-os-config-applier                           |  5 +++
 .../10-os-config-applier                           | 19 ++++++++
 .../install.d/10-os-collect-config                 | 51 ----------------------
 .../10-os-collect-config                           |  5 +++
 .../10-os-collect-config                           | 51 ++++++++++++++++++++++
 .../install.d/01-os-refresh-config                 | 20 ---------
 .../01-os-refresh-config                           |  5 +++
 .../01-os-refresh-config                           | 20 +++++++++
 9 files changed, 105 insertions(+), 90 deletions(-)
 delete mode 100755 elements/os-apply-config/install.d/10-os-config-applier
 create mode 100755 elements/os-apply-config/install.d/os-apply-config-package-install/10-os-config-applier
 create mode 100755 elements/os-apply-config/install.d/os-apply-config-source-install/10-os-config-applier
 delete mode 100755 elements/os-collect-config/install.d/10-os-collect-config
 create mode 100644 elements/os-collect-config/install.d/os-collect-config-package-install/10-os-collect-config
 create mode 100755 elements/os-collect-config/install.d/os-collect-config-source-install/10-os-collect-config
 delete mode 100755 elements/os-refresh-config/install.d/01-os-refresh-config
 create mode 100755 elements/os-refresh-config/install.d/os-refresh-config-package-install/01-os-refresh-config
 create mode 100755 elements/os-refresh-config/install.d/os-refresh-config-source-install/01-os-refresh-config

diff --git a/elements/os-apply-config/install.d/10-os-config-applier b/elements/os-apply-config/install.d/10-os-config-applier
deleted file mode 100755
index e8c01ff..0000000
--- a/elements/os-apply-config/install.d/10-os-config-applier
+++ /dev/null
@@ -1,19 +0,0 @@
-#!/bin/bash
-set -eux
-
-install-packages python-virtualenv
-
-virtualenv --setuptools /opt/stack/venvs/os-apply-config
-
-# bug #1201253 : virtualenv-1.10.1 embeds setuptools-0.9.8, which
-# doesn't manage correctly HTTPS sockets when downloading pbr from
-# https://pypi.python.org/simple/ if using http_proxy and https_proxy
-# envvars
-/opt/stack/venvs/os-apply-config/bin/pip install -U 'setuptools>=1.0'
-/opt/stack/venvs/os-apply-config/bin/pip install -U os-apply-config
-
-ln -s /opt/stack/venvs/os-apply-config/bin/os-apply-config /usr/local/bin/os-apply-config
-ln -s /opt/stack/venvs/os-apply-config/bin/os-config-applier /usr/local/bin/os-config-applier
-
-TEMPLATE_ROOT=$(os-apply-config --print-templates)
-mkdir -p $TEMPLATE_ROOT
diff --git a/elements/os-apply-config/install.d/os-apply-config-package-install/10-os-config-applier b/elements/os-apply-config/install.d/os-apply-config-package-install/10-os-config-applier
new file mode 100755
index 0000000..920512f
--- /dev/null
+++ b/elements/os-apply-config/install.d/os-apply-config-package-install/10-os-config-applier
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+set -eux
+
+install-packages os-apply-config
diff --git a/elements/os-apply-config/install.d/os-apply-config-source-install/10-os-config-applier b/elements/os-apply-config/install.d/os-apply-config-source-install/10-os-config-applier
new file mode 100755
index 0000000..e8c01ff
--- /dev/null
+++ b/elements/os-apply-config/install.d/os-apply-config-source-install/10-os-config-applier
@@ -0,0 +1,19 @@
+#!/bin/bash
+set -eux
+
+install-packages python-virtualenv
+
+virtualenv --setuptools /opt/stack/venvs/os-apply-config
+
+# bug #1201253 : virtualenv-1.10.1 embeds setuptools-0.9.8, which
+# doesn't manage correctly HTTPS sockets when downloading pbr from
+# https://pypi.python.org/simple/ if using http_proxy and https_proxy
+# envvars
+/opt/stack/venvs/os-apply-config/bin/pip install -U 'setuptools>=1.0'
+/opt/stack/venvs/os-apply-config/bin/pip install -U os-apply-config
+
+ln -s /opt/stack/venvs/os-apply-config/bin/os-apply-config /usr/local/bin/os-apply-config
+ln -s /opt/stack/venvs/os-apply-config/bin/os-config-applier /usr/local/bin/os-config-applier
+
+TEMPLATE_ROOT=$(os-apply-config --print-templates)
+mkdir -p $TEMPLATE_ROOT
diff --git a/elements/os-collect-config/install.d/10-os-collect-config b/elements/os-collect-config/install.d/10-os-collect-config
deleted file mode 100755
index db231e5..0000000
--- a/elements/os-collect-config/install.d/10-os-collect-config
+++ /dev/null
@@ -1,51 +0,0 @@
-#!/bin/bash
-set -eux
-
-install-packages python-virtualenv build-essential libz-dev libxslt-dev libxml2-dev python-dev
-
-virtualenv --setuptools /opt/stack/venvs/os-collect-config
-
-# Need setuptools>=1.0 to manage connections when
-# downloading from pypi using http_proxy and https_proxy
-/opt/stack/venvs/os-collect-config/bin/pip install -U 'setuptools>=1.0'
-/opt/stack/venvs/os-collect-config/bin/pip install -U os-collect-config
-
-ln -s /opt/stack/venvs/os-collect-config/bin/os-collect-config /usr/local/bin/os-collect-config
-
-# Minimal static config for bootstrapping
-cat > /etc/os-collect-config.conf <<eof
-[DEFAULT]
-command=os-refresh-config
-eof
-chmod 600 /etc/os-collect-config.conf
-
-if [ "$DIB_INIT_SYSTEM" == "upstart" ] ; then
-    cat > /etc/init/os-collect-config.conf <<eof
-start on runlevel [2345]
-stop on runlevel [016]
-respawn
-exec os-collect-config
-eof
-
-elif [ "$DIB_INIT_SYSTEM" == "systemd" ] ; then
-    cat > /lib/systemd/system/os-collect-config.service <<eof
-[Unit]
-Description=Collect metadata and run hook commands.
-After=cloud-final.service
-Before=crond.service
-
-[Service]
-ExecStart=/usr/local/bin/os-collect-config
-Restart=on-failure
-
-[Install]
-WantedBy=multi-user.target
-eof
-
-  # Enable the service
-  systemctl enable os-collect-config.service
-
-else
-    echo Only systems with systemd or upstart are supported.
-    exit 1
-fi
diff --git a/elements/os-collect-config/install.d/os-collect-config-package-install/10-os-collect-config b/elements/os-collect-config/install.d/os-collect-config-package-install/10-os-collect-config
new file mode 100644
index 0000000..f5d9e0c
--- /dev/null
+++ b/elements/os-collect-config/install.d/os-collect-config-package-install/10-os-collect-config
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+set -eux
+
+install-packages os-collect-config
diff --git a/elements/os-collect-config/install.d/os-collect-config-source-install/10-os-collect-config b/elements/os-collect-config/install.d/os-collect-config-source-install/10-os-collect-config
new file mode 100755
index 0000000..db231e5
--- /dev/null
+++ b/elements/os-collect-config/install.d/os-collect-config-source-install/10-os-collect-config
@@ -0,0 +1,51 @@
+#!/bin/bash
+set -eux
+
+install-packages python-virtualenv build-essential libz-dev libxslt-dev libxml2-dev python-dev
+
+virtualenv --setuptools /opt/stack/venvs/os-collect-config
+
+# Need setuptools>=1.0 to manage connections when
+# downloading from pypi using http_proxy and https_proxy
+/opt/stack/venvs/os-collect-config/bin/pip install -U 'setuptools>=1.0'
+/opt/stack/venvs/os-collect-config/bin/pip install -U os-collect-config
+
+ln -s /opt/stack/venvs/os-collect-config/bin/os-collect-config /usr/local/bin/os-collect-config
+
+# Minimal static config for bootstrapping
+cat > /etc/os-collect-config.conf <<eof
+[DEFAULT]
+command=os-refresh-config
+eof
+chmod 600 /etc/os-collect-config.conf
+
+if [ "$DIB_INIT_SYSTEM" == "upstart" ] ; then
+    cat > /etc/init/os-collect-config.conf <<eof
+start on runlevel [2345]
+stop on runlevel [016]
+respawn
+exec os-collect-config
+eof
+
+elif [ "$DIB_INIT_SYSTEM" == "systemd" ] ; then
+    cat > /lib/systemd/system/os-collect-config.service <<eof
+[Unit]
+Description=Collect metadata and run hook commands.
+After=cloud-final.service
+Before=crond.service
+
+[Service]
+ExecStart=/usr/local/bin/os-collect-config
+Restart=on-failure
+
+[Install]
+WantedBy=multi-user.target
+eof
+
+  # Enable the service
+  systemctl enable os-collect-config.service
+
+else
+    echo Only systems with systemd or upstart are supported.
+    exit 1
+fi
diff --git a/elements/os-refresh-config/install.d/01-os-refresh-config b/elements/os-refresh-config/install.d/01-os-refresh-config
deleted file mode 100755
index 02ee564..0000000
--- a/elements/os-refresh-config/install.d/01-os-refresh-config
+++ /dev/null
@@ -1,20 +0,0 @@
-#!/bin/bash
-# We need to install this early in install.d because other elements will
-# need to use os-refresh-config --print-base to know where to put files
-
-set -eux
-
-install-packages python-virtualenv
-
-virtualenv --setuptools /opt/stack/venvs/os-refresh-config
-
-# Need setuptools>=1.0 to manage connections when
-# downloading from pypi using http_proxy and https_proxy
-/opt/stack/venvs/os-refresh-config/bin/pip install -U 'setuptools>=1.0'
-/opt/stack/venvs/os-refresh-config/bin/pip install -U os-refresh-config
-
-ln -s /opt/stack/venvs/os-refresh-config/bin/os-refresh-config /usr/local/bin/os-refresh-config
-
-for d in pre-configure.d configure.d migration.d post-configure.d; do
-  install -m 0755 -o root -g root -d /opt/stack/os-config-refresh/$d
-done
diff --git a/elements/os-refresh-config/install.d/os-refresh-config-package-install/01-os-refresh-config b/elements/os-refresh-config/install.d/os-refresh-config-package-install/01-os-refresh-config
new file mode 100755
index 0000000..3541947
--- /dev/null
+++ b/elements/os-refresh-config/install.d/os-refresh-config-package-install/01-os-refresh-config
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+set -eux
+
+install-packages os-refresh-config
diff --git a/elements/os-refresh-config/install.d/os-refresh-config-source-install/01-os-refresh-config b/elements/os-refresh-config/install.d/os-refresh-config-source-install/01-os-refresh-config
new file mode 100755
index 0000000..02ee564
--- /dev/null
+++ b/elements/os-refresh-config/install.d/os-refresh-config-source-install/01-os-refresh-config
@@ -0,0 +1,20 @@
+#!/bin/bash
+# We need to install this early in install.d because other elements will
+# need to use os-refresh-config --print-base to know where to put files
+
+set -eux
+
+install-packages python-virtualenv
+
+virtualenv --setuptools /opt/stack/venvs/os-refresh-config
+
+# Need setuptools>=1.0 to manage connections when
+# downloading from pypi using http_proxy and https_proxy
+/opt/stack/venvs/os-refresh-config/bin/pip install -U 'setuptools>=1.0'
+/opt/stack/venvs/os-refresh-config/bin/pip install -U os-refresh-config
+
+ln -s /opt/stack/venvs/os-refresh-config/bin/os-refresh-config /usr/local/bin/os-refresh-config
+
+for d in pre-configure.d configure.d migration.d post-configure.d; do
+  install -m 0755 -o root -g root -d /opt/stack/os-config-refresh/$d
+done
-- 
1.8.5.3

